### ISSUE TITLE
[MNG-273] Add ability to handle plugin configuration as a single unit for context-based [scripted] mojos

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#2688]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** opened **[MNG-273](https://issues.apache.org/jira/browse/MNG-273?redirect=false)** and commented

Currently, the marmalade-mojo stuff works by having one generic class that extends from o.a.m.p.AbstractPlugin, and is responsible for setting up the context, building the marmalade script, and executing it.

When this script wrapper is converted to the new mojo injection method for setting parameters, we will lose the ability for a script to receive parameters from maven (since the generic class will not have those parameter fields to satisfy every script).

In order to enable scripted mojos going forward, develop a new way of handling the plugin configuration (actually mojo configuration, inheriting from the plugin config) as a whole. This includes optional injection of the whole configuration into a single field in the mojo-script wrapper class, and new methods to make navigation of the configuration easier.

If the configuration object has some coercion executed on it before parameters are normally injected, an analogous process needs to happen before this whole-configuration object is injected into the mojo wrapper.

So, to summarize:

- Develop new "whole-configuration" object type and expression
- Add convenience methods for navigating the "whole-configuration" instance from a script
- Add coercion to the "whole-configuration" object's construction, to mimick the behavior of individually injected parameters.

---
No further details from [MNG-273](https://issues.apache.org/jira/browse/MNG-273?redirect=false)

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

fixed; using map-oriented component composer and component configurator for marmalade components in plexus now.

**Comment by jira-importer** [comment]
**[Brett Porter](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brettporter)** commented

works nicely, reused it for beanshell
